core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -1085,39 +1534,49 @@ public void purge(Executor threadPool, final PurgeListener task) {
             }
          }
       }
+      return entriesToPurge;
+   }
+
+   @Override
+   public CompletionStage<Long> size(IntSet segments) {
+      return Flowable.fromPublisher(publishKeys(segments, null)).count().toCompletionStage();
+   }
 
+   @Override
+   public CompletionStage<Long> approximateSize(IntSet segments) {
+      if (!segmented && segments.size() < ctx.getCache().getCacheConfiguration().clustering().hash().numSegments()) {
+         return Flowable.fromPublisher(publishKeys(segments, null)).count().toCompletionStage();
```

Comment:
```suggestion
         return size(segments);
```

Reply:
Yep... I'm looking forward to the day I can remove `maxEntries` and force `segmented=true` all the time :)